### DISCUSSION TITLE
Best approach for Feedback and Testing?

### DISCUSSION THREAD

**Discussion by andyl** [doorgan/sourceror#7] [General]
Hi @doorgan !

I've made progress with [Rfx](https://github.com/andyl/rfx) and am nearly ready to begin writing Sourceror-based edit functions.

I'm able to run all your tests and demo code 100% fine.  When I apply your demo code to larger examples and real-world code, I run into bugs, possibly because your edit functions were written to handle limited-scope demo cases.

To help isolate each issue, I am organizing Sourceror code into standalone modules each with their own tests. Some examples:
- https://github.com/andyl/rfx/blob/master/lib/rfx/edit/credo/multi_alias1.ex
-  https://github.com/andyl/rfx/blob/master/lib/rfx/edit/credo/multi_alias2.ex
- https://github.com/andyl/rfx/blob/master/lib/rfx/edit/proto/comment_add.ex
- https://github.com/andyl/rfx/blob/master/lib/rfx/edit/proto/comment_del.ex

Note the last two don't use Sourceror (yet) but are there to prototype end-to-end flow and test execution.  I've also created a set of Rfx test-helpers that generate on-demand Elixir projects and files for testing.

My question is: when I run into a Sourceror issue, what is the best way to coordinate with you?  I could:

1) create a failing test in Rfx, and file bug reports on the Sourceror repo
2) submit PRs with failing tests to the Sourceror repo
3) let you submit PRs to the Rfx repo
4) or some other workflow...

My preference would be some combination of 1) and 3), but am open to anything really.  Let me know what you think would work best for you.  ;-)

**Comment by doorgan** [comment]
Hi there!

> possibly because your edit functions were written to handle limited-scope demo cases

Pretty much! I picked the multi-alias example because it was complicated enough to figure out the first iterations, but tests in real-word are the most useful right now :slightly_smiling_face: 

> My question is: when I run into a Sourceror issue, what is the best way to coordinate with you? I could:
> 1. create a failing test in Rfx, and file bug reports on the Sourceror repo
> 2. submit PRs with failing tests to the Sourceror repo
> 3. let you submit PRs to the Rfx repo
> 4. or some other workflow...
> My preference would be some combination of 1) and 3), but am open to anything really.

1 and 3 are fine to me as well, as long as we can have a repro with failing tests I can work on a solution and add the corresponding tests to sourceror :slightly_smiling_face: 

Thanks for trying out Sourceror, I look forward to see how Rfx progresses!

**Reply by andyl** [reply]
OK great! I'll post my first issue to the Sourceror tracker over the next day or 2.  :+1:

**Comment by andyl** [comment]
So I posted my first issue (#8) .  To make things easier to replicate, I added a [setup script](https://github.com/andyl/rfx/blob/master/test/rfx_setup.sh) that you can use to clone the repo and checkout the SHA with the failing test.

Hopefully this makes it simpler to dig into the issue.  Let me know if you find bugs in the setup script or have other ideas to streamline interaction.
